transaction/core/src/membership_proofs/mod.rs

```diff
@@ -207,28 +272,13 @@ pub fn derive_proof_at_index(
             element.hash.clone()
         } else {
             // An internal node that contains `index`.
-            // Recompute its hash from its child ranges.
-            let mid: u64 = (element.range.from + element.range.to) / 2;
-
-            // Left child.
-            let left_child_hash = {
-                let left_child_range = Range::new(element.range.from, mid)?;
-                *derived_elements
-                    .get(&left_child_range)
-                    .expect("Child range should already exist.")
-            };
-
-            // Right child.
-            let right_child_hash = {
-                let right_child_range = Range::new(mid + 1, element.range.to)?;
-                *derived_elements
-                    .get(&right_child_range)
-                    .expect("Child range should already exist.")
-            };
-
-            TxOutMembershipHash::from(hash_nodes(&left_child_hash, &right_child_hash))
+            // This is unexpected, none of the proof elements should cover index.
+            return Err(Error::UnexpectedMembershipElement(element_idx));
```

Comment:
this code goes away because the proof structure no longer contains intermediate values, their values are implied by the others, and so they dont need to be transmitted, nor updated